domain/src/main/java/org/oppia/domain/oppialogger/analytics/AnalyticsController.kt

```diff
@@ -154,4 +173,79 @@ class AnalyticsController @Inject constructor(
       )
       .build()
   }
+
+  /**
+   * Checks network connectivity of the device.
+   * Saves the [eventLog] to the [eventLogStore] in the absence of it.
+   * Uploads to remote service in the presence of it.
+   */
+  private fun checkNetworkAndManageEventLog(eventLog: EventLog) {
+    when (networkConnectionUtil.getCurrentConnectionStatus()) {
+      NONE -> addEventLog(eventLog)
+      else -> eventLogger.logEvent(eventLog)
+    }
+  }
+
+  /**
+   * Adds an event to the storage.
+   * At first, it checks if the size of the store isn't exceeding [eventLogStorageCacheSize]
+   * If the limit is exceeded then the least recent event is removed from the [eventLogStore]
+   * After this, the [eventLog] is added to the store.
+   * */
+  private fun addEventLog(eventLog: EventLog) {
+    eventLogStore.storeDataAsync(updateInMemoryCache = true) { oppiaEventLogs ->
+      val storeSize = oppiaEventLogs.eventLogList.size
+      if (storeSize + 1 > eventLogStorageCacheSize) {
+        val eventLogRemovalIndex = getLeastRecentEventIndex(oppiaEventLogs)
+        if (eventLogRemovalIndex != null) {
+          return@storeDataAsync oppiaEventLogs.toBuilder()
+            .removeEventLog(eventLogRemovalIndex)
+            .addEventLog(eventLog)
+            .build()
+        } else {
+          exceptionLogger.logException(
+            NullPointerException(
+              "Least Recent Event index absent -- EventLogCacheStoreSize is 0"
+            )
+          )
+          consoleLogger.e(
+            "Analytics Controller",
+            "Least Recent Event index absent -- EventLogCacheStoreSize is 0"
+          )
+        }
+      }
+      return@storeDataAsync oppiaEventLogs.toBuilder().addEventLog(eventLog).build()
+    }.invokeOnCompletion {
+      it?.let {
+        consoleLogger.e(
+          "DOMAIN",
```

Comment:
Why use ``Analytics Controller`` above but ``DOMAIN`` here? Prefer to keep all of the error tags in a single file consistent.

Reply:
Okay, got it.

Reply:
Done.